sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/SqlDelightCompiler.kt

```diff
@@ -29,6 +29,7 @@ import java.io.Closeable
 private typealias FileAppender = (fileName: String) -> Appendable
 
 object SqlDelightCompiler {
+  private val escapeRegex = Regex("`.*?`|'.*?'|\\[.*?\\]|\".*?\"")
```

Comment:
Hmm this isn't the most efficient regex, but it should reject in the common negative case fairly quickly and only match one branch in the positive case so I guess it's alright.

Does the use of `replace` below match the whole string automatically? Or do we need start and end matchers in this regex here?

Reply:
We could also be naive and just remove all these 4 characters since it's unlikely someone wants that in their names anyone. 

The replace function will replace all occurrences.

Reply:
Fixed